mne_features/feature_extraction.py

```diff
@@ -360,19 +365,60 @@ class ``FeatureExtractor`` can be used as a step in a Pipeline (see
     :func:`extract_features`
     """
 
-    def __init__(self, sfreq=256., selected_funcs=None, params=None, n_jobs=1,
-                 memory=None):
+    def __init__(self, sfreq=256., selected_funcs=None, params=None,
+                 ch_names=None, n_jobs=1, memory=None):
         """Instantiate a FeatureExtractor object."""
         self.sfreq = sfreq
         self.selected_funcs = selected_funcs
         self.params = params
+        self.ch_names = ch_names
         self.n_jobs = n_jobs
         self.memory = memory
+        self.feature_names = None
 
     def fit(self, X, y=None):
-        """Do not have any effect."""
+        """Get the feature names in format
+        ``<ch_name>__<func_params>__<feature>
+        """
+        # we get the channel names and create fake ones if containing '_'
+        # (used later for parsing)
+        unique_str = 'µ&mne-features&µ'
+        if self.ch_names is None:
+            tmp_ch_names = None
+        else:
+            tmp_ch_names = []
+            for ch_name in self.ch_names:
+                if unique_str in ch_name:
+                    raise ValueError(
+                        f'Channel name {ch_name} contains {unique_str}. '
+                        f'Please use another name.')
+                if '_' in ch_name:
+                    # we replace with a special string that will be replaced
+                    tmp_ch_names.append(ch_name.replace('_', unique_str))
+                else:
+                    tmp_ch_names.append(ch_name)
+
+        # trick: use only the first epoch to get the feature names
+        df = extract_features(
+            X[:1],
+            sfreq=self.sfreq,
+            selected_funcs=self.selected_funcs,
+            funcs_params=self.params,
+            n_jobs=1,
+            ch_names=tmp_ch_names,
+            return_as_df=True,
```

Comment:
what you could do is add a parameter in extract_features (eg `separator`) that would default on "_" to keep behavior unchanged but here would be called with "__". And to anticipate any future behavior change you can raise a DeprecationWarning in extract_features in case "_" is passed. Something like this:

separator "_" in feature names will change to "__" in version XXX. You can pass separator="__" to silence this warning.

Reply:
I updated the code to match your request. I also had to include the `separator` parameter in `FeaturesExtractor` to get a consistent separator between the one from `extract_features` (between `ch_name` and `func_params`) and the one from `FeaturesExtractor.fit()` (between `func_params`/`ch_name` and `feature`).
Now the output of both `FeaturesExtractor` and `extract_features` are consistent, keeping the actual behavior and raising a `DeprecatedWarning` for a change to be done in v0.4.
Please tell me if that part is also correct as I have never done that before. (especially the doc)
Thanks for all the feedbacks